operator/controllers/resources/utils.go

```diff
@@ -94,3 +98,29 @@ func getReconciliationCondition(err error) metav1.Condition {
 
 	return condition
 }
+
+func getValidStructureCondition(err error) metav1.Condition {
+	var condition metav1.Condition
+	if err == nil {
+		condition = metav1.Condition{
+			Type:    ConditionTypeValidStructure,
+			Status:  metav1.ConditionTrue,
+			Reason:  ConditionReasonNoError,
+			Message: "Kubernetes CR was successfully decoded.",
+		}
+	} else {
+		condition = metav1.Condition{
+			Type:    ConditionTypeValidStructure,
+			Status:  metav1.ConditionFalse,
+			Reason:  ConditionReasonFailedToDecode,
+			Message: fmt.Sprintf("Failed to decode Kubernetes CR: %s", err),
+		}
+	}
+	return condition
+}
+
+func getUnstructuredObjectFromGVK(gvk schema.GroupVersionKind) *unstructured.Unstructured {
```

Comment:
Move this and getValidStructureCondition to role_controller.go? They don't seem to be used in other files.

Reply:
That sounds OK. Thanks.